ui/raidboss/raidemulator.ts

```diff
@@ -67,6 +72,108 @@ const hideModal = (selector = '.modal.show'): HTMLElement => {
+    defaultLocale(lang, emulatorTooltips.hideSkippedTooltip);
+  querySelectorSafe(document, 'label[for=hideSkipped]').innerHTML =
+    defaultLocale(lang, emulatorTooltips.hideSkipped);
+  querySelectorSafe(document, '.triggerHideCollectorContainer').title =
+    defaultLocale(lang, emulatorTooltips.hideCollectorsTooltip);
+  querySelectorSafe(document, 'label[for=hideCollector]').innerHTML =
+    defaultLocale(lang, emulatorTooltips.hideCollectors);
+
+  // Buttons
+  querySelectorSafe(document, '.loadNetworkLogButton').innerHTML =
+    defaultLocale(lang, emulatorButtons.loadNetworkLog);
+  querySelectorSafe(document, '.exportDBButton').innerHTML =
+    defaultLocale(lang, emulatorButtons.exportDatabase);
+  querySelectorSafe(document, '.importDBButton').innerHTML =
+    defaultLocale(lang, emulatorButtons.importDatabase);
+  querySelectorSafe(document, '.clearDBButton').innerHTML =
+    defaultLocale(lang, emulatorButtons.clearDatabase);
+  querySelectorSafe(document, '.doneButton').innerHTML =
+    defaultLocale(lang, emulatorButtons.done);
+
+  // Repeated buttons
+  querySelectorAllSafe(document, '.yesButton').forEach(
+      (btn) => {
+        btn.innerHTML = defaultLocale(lang, emulatorButtons.yes);
+      });
+  querySelectorAllSafe(document, '.noButton').forEach(
+      (btn) => {
+        btn.innerHTML = defaultLocale(lang, emulatorButtons.no);
+      });
+  querySelectorAllSafe(document, '.closeButton').forEach(
+      (btn) => {
+        btn.innerHTML = defaultLocale(lang, emulatorButtons.close);
+      });
+
+  // Intro modal
+  const introModal = querySelectorSafe(document, '.introModal');
+  querySelectorSafe(introModal, '.modal-title').innerHTML =
+    defaultLocale(lang, emulatorIntroModal.title);
+  querySelectorSafe(introModal, '.modal-body').innerHTML =
+    defaultLocale(lang, emulatorIntroModal.body);
+
+  // Import modal
+  const importModal = querySelectorSafe(document, '.importProgressModal');
+  querySelectorSafe(importModal, '.modal-title').innerHTML =
+    defaultLocale(lang, emulatorImportModal.title);
+  querySelectorSafe(importModal, '.modal-body-contents').innerHTML =
+    defaultLocale(lang, emulatorImportModal.body);
+
+  // Delete database modal
+  const deleteModal = querySelectorSafe(document, '.deleteDBModal');
+  querySelectorSafe(deleteModal, '.modal-title').innerHTML =
+    defaultLocale(lang, emulatorDeleteModal.title);
+  querySelectorSafe(deleteModal, '.modal-body').innerHTML =
+    defaultLocale(lang, emulatorDeleteModal.body);
+
+  // Disconnected modal
+  const discModal = querySelectorSafe(document, '.disconnectedModal');
+  querySelectorSafe(discModal, '.modal-title').innerHTML =
+    defaultLocale(lang, emulatorDisconnectedModal.title);
+  querySelectorSafe(discModal, '.modal-body').innerHTML =
+    defaultLocale(lang, emulatorDisconnectedModal.body);
+
+  // Encounter info template
+  const infoDiv = getTemplateChild(document, 'template.encounterInfo');
```

Comment:
There's a lot of repetition here.  What do you think about just using the keys in these various objects as class names, so you could do a `for (const key in emulatorEncoutnerInfo) { querySelectorSafe(infoDiv, `.${key}`).innerHTML = defaultLocale(lang, emulatorEncounterInfo[key]); }` sorta thing?

I think I just wonder if this repetition could be generalized somehow.

Reply:
I looked at doing this but it would only work for this specific area (`encounterInfo`), so I wasn't sure if it really made sense to set this one translation up different to the rest?

Reply:
I don't understand why that would only work here.  The majority are searching inside `document` vs needing to be inside some other element (arguably those could be adjusted in some way).  Also the majority are looking for a class name vs two for labels (arguably those could be class names).  It seems like they are ~mostly all looking inside document for a class name and then replacing innerHTML.

What makes you say that a more general approach would only work for this specific area? I'm mostly trying to ask if there some way to turn repeated code into data + simpler code.

(My general experience is that well-structured data is harder to get wrong, but repeated code is easy to typo and have mistakes.  I'm not saying this does, but more that it makes me wonder if there's a better way to do this.)

Reply:
I'll wait on the discussion on the other thread, I can probably make this work but it's going to involve rewriting a lot of the HTML/class structure and event hooks in the ts.

Reply:
Latest change should fix this. I'll wait for your feedback on this before I tackle the other comment thread.

The template for encounter info is complicated due to CSS selectors not being able to access template child elements directly. I think it may make sense to ditch the template in this case? That would require some significant changes to `EncounterTab` though to accommodate though.